src/js/system/RoutingService.js

```diff
@@ -0,0 +1,113 @@
+import {Route} from 'react-router';
+
+const queue = {};
```

Comment:
Let's rename this to `routeMap` or something alike and use the builtin`Map` object.  #nit